### ISSUE TITLE
Better Lifx-Z support

### ISSUE THREAD

**Issue by saites** [saites/music-blitz#8]
Currently, Lifx strips (Lifx-Z) are just treated like one big light bulb. It might be possible to get more fun variations out of them.

**Comment by ysaliens** [comment]
What about treating a LIFX-Z as a group of lights (each section being a bulb). You might already have the infrastructure to do this and this way it will be very fluid/dynamic.

If not, what about splitting the different sections as different "bulbs"?

**Comment by saites** [comment]
Yes, splitting the strip into "bulbs" it almost exactly what I'd like to do, with a preference on assigning them all the same "group", irrespective of the group the strip itself is in. I would ignore whatever group the strip is in, and instead split its zones into bulbs, and assign all the bulbs the same group name derived from the strip name. There would be an option to "collapse" or "expand" the strip into its sections or treat it as a single bulb (in which case, it's group name would be respected). You would still be able to override the default group name assigned by the app.

[MultiZone](https://lan.developer.lifx.com/v2.0/docs/multizone-messages#section-setcolorzones-501) light messages only support changing contiguous sections of a strip, so this would likely the best way to handle it. A problem is presented if trying to dispatch many simultaneous messages to the sections, since the controller for the strips may simply drop packets if too many are sent to it in rapid succession. People have been asking LIFX to support setting multiple sections at once with different HSBK values, so I hope they may do so. If they did, I'd like to add an "equalizer" mode for it, where individual sections along the strip correspond to frequency bands, where I could update the intensity of each band in a single message. We'll see.

**Comment by ysaliens** [comment]
Well..hopefully you get to the first part soon, I have my strips shipping right now :)

**Comment by ysaliens** [comment]
Got my LIFX-Z...now I see what you meant by single-bulb. Would love multizone support, especially seeing as I have 4m of it :)

**Comment by saites** [comment]
Same -- I've got a long strip of them as well, but only the one controller, so this is probably the next thing on my list to address

**Comment by ysaliens** [comment]
Anything new with this? Loving my LIFX Z except when it comes to music.

**Comment by saites** [comment]
I have been doing some experimentation, and I managed to confirm with Lifx that they will eventually support setting multiple zones in a single message (which will allow must better processing -- but needs to be accounted for somewhat). However, I am not close to making a release related to it yet.

It is going to need UI improvements in order to be useful, and that is probably the biggest thing people have been asking for. I had hoped that the "minimalist" UI design would be easy and preferred, but it does not seem like that is in fact the case.

For the strips specifically, I need the UI to make it clear that you can treat it as one bulb, or divide the strip into "virtual bulbs". I have an idea in mind that will make that easy, and enhance the way you interact with the bulbs in general. That's what I'm going to work towards, but I don't have a specific schedule for it yet.

Thanks for keeping me honest, though :) Your poking here does keep me inspired to move the app up my priority list.
